recipes/ruy/all/conanfile.py

```diff
@@ -0,0 +1,130 @@
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": False,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "3.4",
+            "apple-clang": "5.1",
+        }
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 14)
+
+        minimum_version = self._minimum_compilers_version.get(str(self.settings.compiler), False)
+        if not minimum_version:
+            self.output.warn("Compiler is unknown. Assuming it supports C++14.")
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration("Build requires support for C++14. Minimum version for {} is {}"
+                .format(str(self.settings.compiler), minimum_version))
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("cpuinfo/cci.20201217")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                destination=self._source_subfolder, strip_root=True)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["RUY_MINIMAL_BUILD"] = True
+        self._cmake.configure()
+        return self._cmake
+
+
+    def build(self):
+        # 1. Allow Shared builds
+        tools.replace_in_file(os.path.join(self._source_subfolder, "cmake", "ruy_cc_library.cmake"),
+                              "add_library(${_NAME} STATIC",
+                              "add_library(${_NAME}"
+                              )
+        # 2. Shared builds fail with undefined symbols without this. clog is a library from the cpuinfo
+        tools.replace_in_file(os.path.join(self._source_subfolder, "ruy", "CMakeLists.txt"),
+                              "set(ruy_6_cpuinfo \"cpuinfo\")",
+                              "set(ruy_6_cpuinfo \"cpuinfo;clog\")"
+                              )
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="*.h", dst=os.path.join("include", "ruy"), src=os.path.join(self._source_subfolder, "ruy"))
+        self.copy(pattern="*", dst="lib", src="lib")
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "ruy"
+        self.cpp_info.names["cmake_find_package_multi"] = "ruy"
```

Comment:
I can't find any call to `install` in the cmake scripts.
```suggestion
```

Reply:
It doesn't have install support. I opened a ticket for them asking for it.
I added these lines since this library is mainly used by tensorflow-lite and it is being used as `find_package(ruy)`.
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/CMakeLists.txt#L138

Reply:
When removing these, the default values of `ruy` can still be used.
We don't explicitly add these when upstream of cmake sets them.

Reply:
It is my understanding that we only explicitly set these variables with the same value as the default when that is actually the name of the module.
In this case nor the library or CMake provide a find module, but this library was created by google to support tfl, and it uses it explicitly as `find_package(ruy)`. If eventually they decide to add a find module it will be surely named `findruy.cmake`.
I don't mind removing these lines tho, but I think it opens up the possibility of someone modifying it on a follow up PR.

Reply:
Indeed, when some name becomes "official", a follow-up pr can add it.